Remove SCL macros for hammer-cli-foreman develop

ofedoren opened the pull request:
ASMR (Another SCL Macros Removal).

I'm not sure if I did it right (did it manually this time due to different output from `./add_gem_package`), it was based on the spec we have for hammer core.

Also, I'm now curious how it did work for few Foreman releases, because SCL stuff is here for quite a long time. I'm also afraid now to do the same stuff for all hammer plugins :melting_face:

ekohl commented on packages/foreman/rubygem-hammer_cli_foreman/rubygem-hammer_cli_foreman.spec:
```diff
@@ -22,56 +19,40 @@ URL: https://github.com/theforeman/hammer-cli-foreman
+Requires: ruby
+Requires: ruby(rubygems)
+Requires: rubygem(hammer_cli) >= 3.7.0
+Requires: rubygem(apipie-bindings) >= 0.6.0
+Requires: rubygem(rest-client) >= 1.8.0
+Requires: rubygem(rest-client) < 3.0.0
+Requires: rubygem(jwt) >= 2.2.1
```
Note to self: perhaps these are autogenerated now and can be skipped. If so, the template should be updated.

ekohl commented on packages/foreman/rubygem-hammer_cli_foreman/rubygem-hammer_cli_foreman.spec:
```diff
@@ -12,7 +9,7 @@
 %{!?_root_sysconfdir:%global _root_sysconfdir %{_sysconfdir}}
 %global hammer_confdir %{_root_sysconfdir}/hammer
```
```suggestion
%global hammer_confdir %{_sysconfdir}/hammer
```

ekohl reviewed (commented):
Normally the SCL macros are unset on the distro. However, on Fedora I've seen some things start to fail somehow. I suspect packit sees something similar. Since we don't care about SCL anymore, it's better to remove it because it also cleans up the spec file.

ofedoren commented:
Thanks, updated. I also changed the changelog entry, because build failed with:
```
fatal: [rubygem-hammer_cli_foreman]: FAILED! => changed=false 

[2023-05-26T16:39:21.436Z]   changelog_epoch_version_release: 3.8.0-0.2.pre.develop

[2023-05-26T16:39:21.436Z]   msg: changelog entry missing for 3.8.0-0.1.pre.develop

[2023-05-26T16:39:21.436Z]   specfile_epoch_version_release: 3.8.0-0.1.pre.develop
```

Not sure if that's right, so just a heads up.
